### ISSUE TITLE
[feature] post_install command hook

### ISSUE THREAD

**Issue by daravi** [conan-io/conan#8570]
Hi. The only hook that I have thought would be useful to me is a `post_install` for example:
1. I would be able to remove my project's CMakeCache.txt file or run `cmake --build . --target clean` after a `conan install`
2. I would be able to update my code editor's include directory paths

So I was wondering:
- Is it possible to add custom hooks or to achieve what I want through other methods?
- If not I would like to request a `post_install` hook.

Edit: To clarify, I need to add some commands to be run after install is done for DevOps purposes.

The workflow is as follows:
- Developer locally modifies a dependency and exports the package to local cache
- Developer then re-runs conan install on the consuming project
- Developer then needs to remove CMakeCache.txt and update their IDE's include paths because the paths have changed

This is definitely not a crucial problem. But it would be nice to be able to manage this step with Conan as well if possible.

[✓] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by solvingj** [comment]
Seems reasonable.  I agree that we should look into it.

**Comment by uilianries** [comment]
Not first time I see someone ask for pre/post install hook.

**Comment by uilianries** [comment]
> Is it possible to add custom hooks

Unfortunately not, they are empty methods listed in conan, which can be overridden only.

It's not totally clear to me, do want insert new behavior after building a project, or after installing a pre-built package from a remote? 

>  to achieve what I want through other methods?

You can try post_build and pre_package if are talking about building a package.

But if you want something related to installing a package from remotes, you can use post_download_package

https://docs.conan.io/en/latest/reference/hooks.html

**Comment by solvingj** [comment]
I think I just figured out why this one was tricky.  In spirit, I think the request is something more like:

`post_install_command`

Where the user doesn't want to hook into the `conan install` operation that takes place for each `conanfile.py` in the graph, but rather the completion of the command. 

@daravi is that accurate?

If so, we haven't talked about this use case in terms of hooks, maybe we can.  I think that in that context. the immediate reaction would be that this might be a candidate for "custom conan commands" in Conan 2.0 (once we define what that feature's scope really would be).  Ping @memsharded thoughts?

**Comment by daravi** [comment]
> Where the user doesn't want to hook into the conan install operation that takes place for each conanfile.py in the graph, but rather the completion of the command.

That is correct. Then my understanding of post_** was not correct since i don't want this for every node in the graph and just need to add some commands to be run after install is done for devops purposes.

The workflow is as follows:
- Developer locally modifies a dependency and exports the package to local cache
- Developer then re-runs `conan install` on the consuming project
- Developer then needs to remove CMakeCache.txt and update their IDE's include paths because the paths have changed

This is definitely not a crucial problem. But it would be nice to be able to manage this step with Conan as well if possible.

**Comment by solvingj** [comment]
Ok, in that case, please update the title of your issue to `post_install command hook`, and then add an "edit" section under your original report, clarifying the request. This helps a lot when triaging it later.

**Comment by memsharded** [comment]
I am closing this ticket as not planned
Referring to Conan 1. Conan 2 has introduced some changes:

- no ``install`` pre/post hooks. Hooks are exclusively for some pre-post recipe methods
- New ``deployers`` extensions
- New custom generators extensions
- New Conan custom commands

Please create new tickets for any further question or issue, thanks for the feedback!
